src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const currentState = { ...state };
```

Comment:
The name 'currentState' might be misleading as it does not represent the current state after each action but rather a copy of the initial state. Consider renaming it to 'stateCopy' or something similar to better reflect its purpose.